File: lab6dict.py
from collections import defaultdict

def groupAndSortOwners(file_dict):
    grouped = defaultdict(list)
    
    # Group files by owner
    for file, owner in file_dict.items():
        grouped[owner].append(file)
    
    # Sort the file list for each owner
    for owner in grouped:
        grouped[owner].sort()
    
    return dict(grouped)

files = {
    'Input.txt': 'Albert',
    'Code.py': 'Stan',
    'Output.txt': 'Albert',
    'btech.txt': 'Albert'
}

result = groupAndSortOwners(files)
print(result)

File: lab6dict.java
import java.util.*;

public class lab6dict {

    public static Map<String, List<String>> groupAndSortOwners(Map<String, String> fileMap) {
        Map<String, List<String>> groupedMap = new HashMap<>();

        // Grouping
        for (Map.Entry<String, String> entry : fileMap.entrySet()) {
            String file = entry.getKey();
            String owner = entry.getValue();
            groupedMap.computeIfAbsent(owner, k -> new ArrayList<>()).add(file);
        }

        // Sorting file names alphabetically
        for (List<String> fileList : groupedMap.values()) {
            Collections.sort(fileList);
        }

        return groupedMap;
    }

    public static void main(String[] args) {
        Map<String, String> files = new HashMap<>();
        files.put("Input.txt", "Albert");
        files.put("Code.py", "Stan");
        files.put("Output.txt", "Albert");
        files.put("btech.txt", "Albert");

        Map<String, List<String>> result = groupAndSortOwners(files);

        for (Map.Entry<String, List<String>> entry : result.entrySet()) {
            System.out.println(entry.getKey() + ": " + entry.getValue());
        }
    }
}